### ISSUE TITLE
Start Tracking This Project freezes VS Code (fixed)

### ISSUE THREAD

**Issue by reZach** [markm208/storyteller#5]
**How to reproduce**

VS Code version: 1.24.1
OS: Windows 10

1. Create a new file (ctrl + N)
2. Start Tracking This Project (from VS command pallette, ctrl + shift + p) 
3. After a few seconds the window is unresponsive and needs to reload

It seems to happen if you start tracking this project if the file is not in a folder.

**Comment by markm208** [comment]
I can't seem to reproduce this. When I attempt to start tracking without an open folder I get a message, "_You must open a folder to use Storyteller. You can use Storyteller by selecting 'Storyteller: Start Tracking This Project' from the command palette or clicking the 'Start Storyteller' button below._ "

You have no folder open?

**Comment by reZach** [comment]
I believe I was in another directory and created a file with ctrl + n. This new file was not in a directory as it had not been saved yet. I will try to get more details and comment on this later.

**Comment by markm208** [comment]
Yes, this was a bug. In my testing I always create a new file by using the file system view to the left of the editor window. I never really used the File -> New menu before. I believe I have fixed the issue(a confirmation would be great). I have updated the plugin it should ask you to update next time you start VS Code.

One issue with the implementation of the bug fix is that before a File -> New file has been saved to the open storyteller project folder it will not capture the entire history of the file before the save point. In other words, the contents of the file at the time of the save to the open storyteller project folder will be added to the system all at once. After it has been added then all of the changes after that point will be captured like normal.

Using the file system view next to the editor window is probably the safest way to add new files since it will capture the entire history of the file.
